src/main/java/com/google/sps/data/Places.java

```diff
@@ -156,6 +170,31 @@ public int compare(PlacesSearchResult hiddenGem_1, PlacesSearchResult hiddenGem_
         return Float.compare(hiddenGem_2.rating, hiddenGem_1.rating);
       }
     });
-    return sortedHiddenGems;
+    return removeDuplicates(sortedHiddenGems);
+  }
+
+  /**
+   * This function returns the list of hidden gems without the duplicates.
+   * @param hiddenGems                 This takes as parameter the set of hidden gems to be filtered.
+   * @return List<PlacesSearchResult>  This return a list of hidden gems without the duplicates.
+   */
+  public static List<PlacesSearchResult> removeDuplicates(List<PlacesSearchResult> hiddenGems) {
+    List<PlacesSearchResult> distinctHiddenGems = hiddenGems.stream() 
+      .filter(distinctByKey(hiddenGem -> hiddenGem.placeId))
+      .collect(Collectors.toList());
+      
+    return distinctHiddenGems;
+  }
+  
+  /**
+   * Utility function to filter a list by class field.
+   * Resource used: https://howtodoinjava.com/java8/java-stream-distinct-examples/ 
```

Comment:
What sort of licensing does howtodoinjava have?
w3schools has a license page that says they code on their site can be used for any purpose which is why I've happily let some stuff from there in.
The bottom of hte site says "all rights reserved" which I suspect means we probably can't use it for commercial uses. See if you can have a go at writing something yourself, they're basically putting all the gems in a hashmap keyed on id and pulling out the result. They've done a bunch of work to make it super generic and thread safe but you won't need that for this project.

Reply:
I tried rewriting the function using a HashMap with key as the hidden gem's place ID and it seems to work to remove the duplicates